WcaOnRails/app/webpacker/javascript/edit-schedule/SchedulesEditor/index.js

```diff
@@ -72,28 +67,37 @@ export default class SchedulesEditor extends React.Component {
           />
         </Col>
         <Col xs={9}>
-          <Panel>
-            <Panel.Heading>
-              <RoomSelector
-                scheduleWcif={scheduleWcif}
-                selectedRoom={selectedRoom}
-                handleRoomChange={handleRoomChange}
-              />
-            </Panel.Heading>
-            <Panel.Body>
-              {selectedRoom.length === 0 ? (
-                <div>Please select a room to edit its schedule</div>
-              ) : (
-                <EditScheduleForRoom
-                  scheduleWcif={scheduleWcif}
-                  locale={locale}
-                  keyboardEnabled={keyboardEnabled}
-                  handleKeyboardChange={handleToggleKeyboardEnabled}
-                  selectedRoom={selectedRoom}
-                />
-              )}
-            </Panel.Body>
-          </Panel>
+          {scheduleWcif === null
+          || scheduleWcif.venues === null
+          || scheduleWcif.venues.length === 0 ? (
+            <Alert bsStyle="warning">
+              Please add a venue in the `&apos;`Edit venues information`&apos;`
+              section above
+            </Alert>
+            ) : (
+              <Panel>
+                <Panel.Heading>
+                  <RoomSelector
+                    scheduleWcif={scheduleWcif}
+                    selectedRoom={selectedRoom}
+                    handleRoomChange={handleRoomChange}
+                  />
+                </Panel.Heading>
+                <Panel.Body>
+                  {selectedRoom.length === 0 ? (
+                    <div>Please select a room to edit its schedule</div>
+                  ) : (
+                    <EditScheduleForRoom
```

Comment:
I think the check for the timezone should be performed somewhere here, before throwing in the `EditScheduleForRoom` component.
Right now if there is a room with a valid schedule and you add a room in a new venue, without timezone, and then visit the schedule for that new room, it will display the appropriate message but it will then crash upon returning to the valid room's schedule.
Despite not displaying the actual schedule, emitting the component does fire the react's callbacks, and fullcalendar is initialized.
I'm not exactly sure why it doesn't throw an error right away, but basically events that we request from it afterwards are broken/not there, and the whole thing crash.

Putting the `venueWcif.timezone ? ...` check here instead of within the component seems to work fine so I'd suggest doing that!